tests/unit/browser/test_cache.py

```diff
@@ -35,6 +35,58 @@ def preload_cache(cache, url='http://www.example.com/', content=b'foobar'):
     cache.insert(device)
 
 
+def test_cache_config_change_cache_size(config_stub, tmpdir):
+    """Change cache size and emit signal to trigger on_config_changed."""
+    max_cache_size = 1024
+    config_stub.data = {
+        'storage': {'cache-size': max_cache_size},
+        'general': {'private-browsing': False}
+    }
+    disk_cache = cache.DiskCache(str(tmpdir))
+    assert disk_cache.maximumCacheSize() == max_cache_size
+
+    config_stub.data['storage']['cache-size'] = max_cache_size * 2
+    config_stub.changed.emit('storage', 'cache-size')
```

Comment:
Please use `config_stub.set(sect, opt, value)` instead of emitting the signal manually everywhere - it sets the value and emits the signal.